### ISSUE TITLE
Can't inherit from a model that has history enabled.

### ISSUE THREAD

**Issue by jeffreypriebe** [keystonejs/keystone#1524]
From the inheritance example in the [Keystone docs](http://keystonejs.com/docs/database/), if you turn on the history plugin on the BasePage, you will get an error when trying to launch keystone.
Mongoose throws
```
throw new mongoose.Error.OverwriteModelError(name);
OverwriteModelError: Cannot overwrite `basepages_revisions` model once compiled.
at Mongoose.model (website\node_modules\keystone\node_modules\mongoose\lib\index.js:334:13)
```

As I understand it, this is because the "history" plugin (keystone: lib\schemaPlugins\history.js) creates a model ("basepages_revisions" in the example) whose name is based off the Base model. When it comes around to creating the inherited model, the history plugin tries to create another model - using the same Base information and thus a model with the same name. Mongoose catches the repeat model and throws an error.

See [Gist](https://gist.github.com/jeffreypriebe/d2e3d04303370cf99a00) for example case.

I expected to be able to turn on history, configure inheritance and for it to just work.

### How to fix this?

Note: you can't even disable the history on BasePages and then enable it on ChildPages. The history plugin will still try to write its data into "basepages_revisions."
So, you could get away with this for a single ChildPages (where ChildPages has history, BasePages does not).
But, you cannot create a second "SecondChildPages" inheriting from BasePages with history enabled. Same error as above for the same reason.

**Comment by jeffreypriebe** [comment]
Note sure if [previous closed issue "Plugins seems to break when using List inheritance"](https://github.com/keystonejs/keystone/issues/1363) is related.

**Comment by lojack** [comment]
What version of keystone are you using @jeffreypriebe?

**Comment by jeffreypriebe** [comment]
[v0.3.12 / Jun 25, ecbf4ae](https://github.com/keystonejs/keystone/tree/ecbf4ae309f97a9627390757945c66c33a682410) - current latest on [npm](https://www.npmjs.com/package/keystone).

Have you seen different results with a different version?

**Comment by JedWatson** [comment]
Closed by #1526
